src/audio-manager.ts

```diff
@@ -1,261 +1,389 @@
+                this._sfxGain.gain.linearRampToValueAtTime(volume, time);
+                break;
+            default:
+                this._masterGain.gain.linearRampToValueAtTime(volume, time);
         }
     }
 
     /**
-     * Checks if the audio node is currently playing.
+     * Removes all decoded audio from the manager that is associated with the given ID.
+     *
+     * @warning This will stop the playback of the given ID.
+     * @param id Identifier of the audio that should be removed.
      */
-    get isPlaying(): boolean {
-        return this._isPlaying;
+    remove(id: number) {
+        this.stop(id);
+        this._bufferCache.delete(id);
     }
 
     /**
-     * Sets the volume of this PlayableNode.
+     * Frees a player that was currently playing the given ID.
+     *
+     * @warning This is for internal use only, use at your own risk!
+     * @param id Identifier of previously playing audio.
      */
-    set volume(v: number) {
-        const time = _audioContext.currentTime + RAMP_TIME;
-        this._gainNode.gain.exponentialRampToValueAtTime(v, time);
+    _freeUpBusyPlayer(id: number) {
+        const player = this._busyPlayers.get(id);
+        if (player) {
+            this._busyPlayers.delete(id);
+            this._freePlayers.push(player);
+        }
     }
 
     /**
-     * Free's up the audio resources after Node stopped playing.
+     * Gets the current amount of free regular players in the audio manager.
+     *
+     * @note Use this to check how many resources your current project is actually using, and then optimize the
+     * amount of regular players in the AudioManager constructor.
      */
-    destroy() {
-        if (this._isPlaying) {
-            this._destroy = true;
-        } else {
-            this._audioManager._remove(this._source);
-            this._gainNode.disconnect();
-        }
+    get amountOfFreePlayers() {
+        return this._freePlayers.length;
+    }
 
-        this.destroy = () => {};
+    /**
+     * Checks if a one-shot has been stopped to make room for another.
+     *
+     * @note This could indicate that there aren't enough one-shot players. Keep in mind though, that it might not
+     * be noticeable (and therefore doesn't matter) if the last started one-shot had to stop before its natural end.
+     *
+     * @returns true if all players were in use at one point and a player had to be stopped to make room for another
+     * one-shot.
+     */
+    get hasStoppedOneShot() {
+        return this._hasOneShotStopped;
     }
 }
 
-export const globalAudioManager = new AudioManager();
+// @todo: Question for Timmy: Not sure if we need to give the user a instance or not. What do you think? It would
+//  kind of defeat the purpose of having it configurable in the first place. Or should we come up with reasonable
+//  defaults and make it not configurable at all?
+export const globalAudioManager = window.AudioContext ? new AudioManager() : null;
```

Comment:
This question is the reason it took me so long to finish reviewing. I think it's nice to have both options. If you just want to start a song in a scene, and play a few sounds it is perfect with the provided components. If a game gets more complex with voice lines, timed audio or more complex things it's nice to have the option to use the same audio manager, but use it in your own way. That's probably how I would use it like this in my games. I don't think I would ever use other constructors.

Reply:
Alright, so we just provide the `globalAudioManager` and keep it at 16 one-shots and 16 regular players?